### PR TITLE
[LoRA] Handle DoRA better

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Our support for loading community DoRA checkpoints is currently broken. See https://github.com/KohakuBlueleaf/LyCORIS/issues/216. 

So, this PR:

1. Detects if a state dict has `dora_scale` present. Upon detection, it logs a warning about the incompatibility and filters the state dict so that it doesn't have the `dora_scale` keys. 
2. Adjust the tests accordingly. 

Once Ben gives a go ahead I will request for a review from Yiyi.

### CODE REVIEW

**Review by BenjaminBossan — APPROVED** [review]
Yes, I agree it's better not to load these DoRA weights, as we now pretty much agree with Kohaku that their DoRA implementation is using the other axis.

Just a nit, the unit test could check for the new warning message.

**Comment by sayakpaul** [comment]
@BenjaminBossan thanks. Does https://github.com/huggingface/diffusers/pull/9547/commits/a8fdc76865824cb2b9e6d0e4db4a4a5187d2fb6d work for you? 

@yiyixuxu could you give this a look too?

**Review by BenjaminBossan — APPROVED** [review]
Thanks, LGTM.

File: src/diffusers/loaders/lora_pipeline.py [hunk 1]
```diff
@@ -1125,7 +1136,13 @@ def load_lora_weights(
         # First, ensure that the checkpoint is a compatible one and can be successfully loaded.
         state_dict = self.lora_state_dict(pretrained_model_name_or_path_or_dict, **kwargs)
 
-        is_correct_format = all("lora" in key or "dora_scale" in key for key in state_dict.keys())
+        is_dora_scale_present = any("dora_scale" in k for k in state_dict)
+        if is_dora_scale_present:
```
Comment (yiyixuxu) [inline src/diffusers/loaders/lora_pipeline.py:1140]: I see that we sometimes add this code to `lora_state_dict()` and sometimes on `load_lora_weights`. Is there any reason we do this differently?

Comment (sayakpaul) [reply src/diffusers/loaders/lora_pipeline.py:1140]: Sure.

It's because of the way `lora_state_dict()` and `load_lora_weights()` are used. 

`load_lora_weights()` is often the entry point to our users for loading a LoRA weight to perform inference. However, `lora_state_dict()` is also used during training, especially for resuming training from an existing checkpoint. Example:

https://github.com/huggingface/diffusers/blob/1287822973e87fc8739a608b7296fe1797b1b074/examples/dreambooth/train_dreambooth_lora_sdxl.py#L1264

LMK if that makes sense.

Comment (sayakpaul) [reply src/diffusers/loaders/lora_pipeline.py]: In e08cf7442, I have tried to make this more consistent by moving all the "dora_scale" check related code to `lora_state_dict()`. 

LMK if that works for you, @yiyixuxu.

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by sayakpaul** [comment]
Failing test is unrelated.
